### PR TITLE
Allow PHP 8

### PR DESCRIPTION
**Pull request by simPod**

### CODE REVIEW

**Comment by simPod** [comment]
Build failure is because of deprecation warning in `libxml_disable_entity_loader`. 

Fixed in https://github.com/schmittjoh/serializer/pull/1267.

**Comment by goetas** [comment]
hello! I'm fine with this changes except for `drop EOLed PHP versions`. Is there any good reason to do so?

**Comment by simPod** [comment]
It was because of phpunit I think. Will look into it.

**Comment by Yozhef** [comment]
@goetas Any news?

File: DependencyInjection/JMSSerializerExtension.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ public function loadInternal(array $config, ContainerBuilder $container)
             $container->setParameter('jms_serializer.infer_types_from_doctrine_metadata', false);
         }
 
-        if (PHP_VERSION_ID >= 70400 && class_exists(TypedPropertiesDriver::class)) {
+        if (class_exists(TypedPropertiesDriver::class)) {
```
Comment (goetas) [inline DependencyInjection/JMSSerializerExtension.php:111]: why this change? we do not want to enable `TypedPropertiesDriver` for 7.2 or 7.4

**Review by goetas — CHANGES_REQUESTED** [review]

File: DependencyInjection/JMSSerializerExtension.php (hunk 1 above)
Comment (simPod) [reply DependencyInjection/JMSSerializerExtension.php:111]: I wish I remembered. Will revert and see what happens

File: .travis.yml [hunk 2]
```diff
@@ -22,7 +23,7 @@ matrix:
 
 before_script:
   - if [[ "$SYMFONY_VERSION" != "" ]]; then composer require --dev --no-update symfony/symfony=$SYMFONY_VERSION; fi
-  - COMPOSER_MEMORY_LIMIT=-1 composer update ${COMPOSER_FLAGS} --no-interaction
+  - COMPOSER_MEMORY_LIMIT=-1 composer update ${COMPOSER_FLAGS} --no-interaction if [[ "$(phpenv version-name)" == "8.0snapshot" ]]; then --ignore-platform-req=php; fi
```
Comment (jacekpl) [inline .travis.yml:26]:
```
COMPOSER_MEMORY_LIMIT=-1 composer update ${COMPOSER_FLAGS} --no-interaction $(if [[ "$(phpenv version-name)" == "8.0snapshot" ]]; then --ignore-platform-req=php; fi)
```

Comment (simPod) [reply .travis.yml:26]: what would it do?

**Comment by goetas** [comment]
Done in https://github.com/schmittjoh/JMSSerializerBundle/pull/830
